DataManagementSystem/DB/FileCatalogComponents/FileMetadata.py

```diff
@@ -13,442 +13,486 @@
       fileID = result['Value']['Successful'][path]['FileID']
     else:
-      return S_ERROR( 'File %s not found' % path )
+      return S_ERROR('File %s not found' % path)
 
     failedMeta = {}
     for meta in metadata:
       if meta in metaFields:
         # Indexed meta case
-        req = "DELETE FROM FC_FileMeta_%s WHERE FileID=%d" % ( meta, fileID )
-        result = self.db._update( req )
+        req = "DELETE FROM FC_FileMeta_%s WHERE FileID=%d" % (meta, fileID)
+        result = self.db._update(req)
         if not result['OK']:
           failedMeta[meta] = result['Value']
       else:
         # Meta parameter case
-        req = "DELETE FROM FC_FileMeta WHERE MetaKey='%s' AND FileID=%d" % ( meta, fileID )
-        result = self.db._update( req )
+        req = "DELETE FROM FC_FileMeta WHERE MetaKey='%s' AND FileID=%d" % (meta, fileID)
+        result = self.db._update(req)
         if not result['OK']:
           failedMeta[meta] = result['Value']
 
     if failedMeta:
       metaExample = failedMeta.keys()[0]
-      result = S_ERROR( 'Failed to remove %d metadata, e.g. %s' % ( len( failedMeta ), failedMeta[metaExample] ) )
+      result = S_ERROR('Failed to remove %d metadata, e.g. %s' % (len(failedMeta), failedMeta[metaExample]))
       result['FailedMetadata'] = failedMeta
     else:
       return S_OK()
 
-  def __getFileID( self, path ):
+  def __getFileID(self, path):
 
-    result = self.db.fileManager._findFiles( [path] )
+    result = self.db.fileManager._findFiles([path])
     if not result['OK']:
       return result
     if result['Value']['Successful']:
       fileID = result['Value']['Successful'][path]['FileID']
     else:
-      return S_ERROR( 'File not found' )
-    return S_OK( fileID )
+      return S_ERROR('File not found')
+    return S_OK(fileID)
 
-  def __setFileMetaParameter( self, fileID, metaName, metaValue, credDict ):
+  def __setFileMetaParameter(self, fileID, metaName, metaValue, credDict):
     """ Set an meta parameter - metadata which is not used in the the data
-        search operations
+        search operations.
+        Modified by JM: make metaName Dirac group aware.
     """
-    result = self.db._insert( 'FC_FileMeta',
-                          ['FileID', 'MetaKey', 'MetaValue'],
-                          [fileID, metaName, str( metaValue )] )
+    result = self.db._insert('FC_FileMeta',
+                             ['FileID', 'MetaKey', 'MetaValue'],
+                             [fileID, self._getMetaName(metaName, credDict), str(metaValue)])
     return result
 
-  def setFileMetaParameter( self, path, metaName, metaValue, credDict ):
+  def _getMetaName(self, meta, credDict):
+    """
+    Return a metadata name based on client supplied meta name and client credentials
+    For the base class it just returns the name passed in.
+    This method is a 'hook' and is meant to be overwritten by derived classes.
+
+    :param meta:  meta name
+    :param credDict: client credentials
```

Comment:
```
    :param str meta:  meta name
    :param dict credDict: client credentials
```
(also later)

Reply:
These have been addressed in the last push. (mix-in)